### DISCUSSION TITLE
Figure out how to preserve behavior of WebView2 when uninstalling Edge

### DISCUSSION THREAD

**Discussion by nopeless** [LeDragoX/Win-Debloat-Tools#123] [Ideas]
It appears that WebView2 is necessary for certain tools and applications

Visual Studio Community version when installing certain components require WebView2

Notable ones include PWA apps such as Snapchat and Instagram. They straight out fail due to "unknown reasons" but is easily fixable by reinstalling Edge

@LeDragoX Can you replicate this behavior?

Stock windows -> Remove Edge -> Install any PWA app from ms store

**Comment by LeDragoX** [comment]
I'm using Hyper-V to create a Windows 11 Pro install. Sorry, I only had a portuguese ISO, but shouldn't be a problem.

# Edge Uninstalled

- Logged out the user and entered again.
- Edge shortcuts and in Control panel weren't available anymore.
- Proceeded to install Instagram App and failed.
![image](https://github.com/LeDragoX/Win-Debloat-Tools/assets/27643047/276ebbf5-4678-4730-9866-1a46f5290de0)

Couldn't install Edge via script GUI.

# Force updated the apps

- This was needed as `Winget` wasn't working, then ended updating **App Installer**.
![image](https://github.com/LeDragoX/Win-Debloat-Tools/assets/27643047/cb38fe39-ae37-49b9-b6e6-4dfdbbd23a8f)

- Installed Firefox through GUI
- Downloaded Edge manually from: https://www.microsoft.com/en-us/edge/download?form=MA13FJ
- Edge is working once again

# Was able to install Instagram again.
![image](https://github.com/LeDragoX/Win-Debloat-Tools/assets/27643047/4cbaa433-35df-456f-9993-ca16a70645b6)

**Reply by nopeless** [reply]
Shouldn't it be possible to uninstall edge but still use webview2?

**Reply by LeDragoX** [reply]
Yes, it's possible, they are 2 components aparted from each other. I'll do this.

**Comment by ulises2k** [comment]
The Instagram application is a "Fake". It's just a browser link to go to the official website. The Instagram app is just the Edge browser with the Instagram website.

**Reply by nopeless** [reply]
you mean its a PWA? not sure what you mean by fake

either way, if thats how ms store implements web apps i believe you need to mention it somewhere

**Comment by LeDragoX** [comment]
![image](https://github.com/LeDragoX/Win-Debloat-Tools/assets/27643047/a3c723ef-6558-474a-8aba-d315e31de96f)
**Widgets** still working, but had to keep the **EdgeCore** and **EdgeWebView** on `C:\Program Files (x86)\Microsoft\`.

Still unable to install **Instagram/Facebook** (AKA PWA apps), as it requires Edge to be installed, this was a headache to do, but finally with some different results.

**Reply by nopeless** [reply]
thank you for investigating, keep me updated

**Reply by LeDragoX** [reply]
I tried my best to find a way to keep EdgeWebView2, but it's not possible, because the moment you uninstall Edge, it breaks its connection with WebView2, but at least keeping it's files from **C:\Program Files (x86)\Microsoft\EdgeCore** and **EdgeWebView**, didn't broke **Widgets** on taskbar. I want to merge the commits til' this moment.

**Reply by nopeless** [reply]
understandable, how about disabling Edge update schedules, removing startup task, and removing it from start menu then? Additionally, what do you think about keeping the original "remove edge" button, but with a red background and a note that says ("will uninstall WebView2 and thus break many PWA applications")?
